### PR TITLE
Inheritance 

### PR DESCRIPTION
**Pull request by stellakunzang**
Added Collection parent class
Eliminated repetition by moving overlapping function in Game, GameTeam, and Team classes to the Collection parent class.

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -1,29 +1,10 @@
...
-  end
-#inheritance, more general? 2 arguments passed
-  def find_by(id)
-   all.find_all{|game| game.game_id==id}
-  end
-
-  def self.find_by(id)
-   all.find_all{|game| game.game_id==id}
-  end
-  #deliverable
```
Comment (stellakunzang) [inline lib/game.rb:26]: Deleting code, YaY!

File: lib/game.rb [hunk 2]
```diff
@@ -38,10 +19,10 @@ def self.grouped_by_season(passed_in_season)
 
   def self.count_of_games_by_season
     # this can be refactored to include ross' games_per(:season) method -sb
-    games_by_season = @@all.group_by { |game| game.season }
+    games_by_season = all.group_by { |game| game.season }
     count = {}
     games_by_season.keys.each do |key|
-      count[key.to_s] = @@all.count { |game| game.season == key}
```
Comment (stellakunzang) [inline lib/game.rb:44]: caught a few of these stragglers

File: lib/team.rb [hunk 3]
```diff
@@ -20,24 +14,14 @@ def self.find_team_info(id)
...
   def self.count_of_teams
     all.length
   end
 
-  attr_reader :game_teams, :team_id, :franchise_id, :team_name, :abbreviation, :stadium, :link, :team_name
+  attr_reader :team_id, :franchise_id, :team_name, :abbreviation, :stadium, :link, :team_name
 
 
   def initialize(team_info)
-    @game_teams = game_teams
```
Comment (stellakunzang) [inline lib/team.rb:40]: it appeared that this wasn't being utilized at all. Deleted and caused no errors or concern.

Comment (foleymichelle9) [reply lib/team.rb:40]: that was probably from when my methods were in team class! Absolutely not needed haha

File: lib/game.rb (hunk 1 above)
Comment (foleymichelle9) [reply lib/game.rb:26]: nice!

**Comment by foleymichelle9** [comment]
This looks great Stella! I'm so excited we got it to work! I approve merging

File: lib/collection.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+require "csv"
+class Collection
+
```
Comment (perryr16) [inline lib/collection.rb:3]: very excited about this

File: lib/game.rb (hunk 1 above)
Comment (perryr16) [reply lib/game.rb:26]: yay!

File: lib/game.rb [hunk 5]
```diff
@@ -78,7 +59,7 @@ def self.games_goals_by(hoa_team)
   end
 #deliverable
   def self.average_goals_by(hoa_team)
-      divide_hash_values(:goals, :games_played, games_goals_by(hoa_team))
+    divide_hash_values(:goals, :games_played, games_goals_by(hoa_team))
   end
```
Comment (perryr16) [inline lib/game.rb:63]: subtle

**Review by perryr16 — COMMENTED** [review]
This is great -rp

File: lib/game.rb (hunk 5 above)
Comment (stellakunzang) [reply lib/game.rb:63]: OCD can't help fixing indentation

**Comment by Rostammahabadi** [comment]
Merge conflict but it looks great!

**Pull request merged**